src/tasks/withdraw/safe.ts

```diff
@@ -0,0 +1,59 @@
+import SafeApiKit from "@safe-global/api-kit";
+import Safe, { EthersAdapter } from "@safe-global/protocol-kit";
+import { SafeTransactionDataPartial } from "@safe-global/safe-core-sdk-types";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+
+interface Transaction {
+  authoringSafe: string;
+  to: string;
+  data: string;
+}
+
+function serviceUrlForNetwork(network: string): string {
+  if (["goerli", "mainnet", "gnosis-chain"].includes(network)) {
+    return `https://safe-transaction-${network}.safe.global`;
+  } else if (network === "xdai") {
+    return "https://safe-transaction-gnosis-chain.safe.global/";
+  } else {
+    throw new Error(`Unsupported network ${network}`);
+  }
+}
+
+// Creates and proposes a transaction to the Safe Multisig, which can then be confirmed by other signers in the Web UI. Returns the link to the transaction in the Web UI.
+export async function proposeTransaction(
+  { ethers }: HardhatRuntimeEnvironment,
+  network: string,
+  { authoringSafe, to, data }: Transaction,
+): Promise<string> {
+  const proposer = (await ethers.getSigners())[0];
```

Comment:
Comment: I find the following slightly more aesthetically pleasing:
```suggestion
  const [proposer] = await ethers.getSigners();
```